boggle.js

```diff
@@ -15,17 +20,56 @@ function printBoggleBoard(boggleBoard) {
   }
 }
 
+// function shuffle(allArrays) {
+//   for (let a = 0; a <= 7; a++) {
+//     let randomDie = Math.floor(Math.random() * 16);
+//     allArrays.push(allArrays[randomDie]);
+//     allArrays.splice(randomDie, 1);
+//   }
+//   console.log(allArrays);
+// }
+
+function randomRange(length) {
+  return Math.floor(Math.random() * length - 1);
+}
+
+function removeElement(allArrays, index) {
+  let item = allArrays.splice(index, 1)[0];
+  return item;
+}
+
+function shuffle(allArrays) {
+  while (allArrays.length > 0) {
+    let randIndex = randomRange(allArrays.length);
+    let removedElem = removeElement(allArrays, randIndex);
+    shuffledArray.push(removedElem);
+   // return shuffledArray;
+    // console.log(removedElem);
+  }
+  return shuffledArray;
+}
 
 function shake(board) {
- for (a=0; a<=15; a++){
-   board[a] = alphabet[Math.floor(Math.random()*26)];
- }
+  let finalArray = shuffle(allArrays);
+  console.log(finalArray);
+  let allArraysIndex = 0;
+  for (let a = 0; a < 4; a++) {
```

Comment:
What is the board were a different size? Don't hard code numbers here. You know how to find the length of an array.